programs/sig-aggregator/src/attestor_data.rs

```diff
@@ -16,76 +14,163 @@ type Signature = FixedBytes<SIGNATURE_BYTE_LENGTH>;
 pub const PUBKEY_BYTE_LENGTH: usize = 33;
 type Pubkey = FixedBytes<PUBKEY_BYTE_LENGTH>;
 
-//  HashMap<height, HashMap<State, Vec[(Signatures, pub_key)]>>
-//  Height: 101
-//      State: 0x1234... (32 bytes)
-//          Sign_PK: [(SigAtt_A, PK_Att_A), (SigAtt_B, PK_Att_B)]
-//      State: 0x9876...
-//          Sign_PK: [(SigAtt_C, PK_Att_C), (SigAtt_D, PK_Att_D), (SigAtt_E, PK_Att_E)]
-//  Height: 102
-//      State: 0x5678...
-//          Sign_PK: [(SigAtt_A, PK_Att_A), (SigAtt_B, PK_Att_B), (SigAtt_C, PK_Att_C), (SigAtt_D, PK_Att_D), (SigAtt_E, PK_Att_E)]
-type SignedStates = HashMap<State, Vec<(Signature, Pubkey)>>;
+/// Maps attested_data -> list of attestations
+///
+/// Structure:
+/// ```text
+/// Attested_data: 0x1234... (12 bytes)
+///     [Attestation_A, Attestation_B]
+/// Attested_data: 0x9876...
+///     [Attestation_C, Attestation_D]
+/// ```
 
-#[derive(Debug, Clone)]
-pub struct AttestatorData(HashMap<Height, SignedStates>);
+#[derive(Debug, Clone, Default)]
+pub struct AttestatorData {
+    state_attestations: HashMap<State, Vec<Attestation>>,
+}
 
 impl AttestatorData {
+    #[must_use]
     pub fn new() -> Self {
-        Self(HashMap::new())
+        Self::default()
     }
 
-    pub fn insert(&mut self, att_resp: AttestationsFromHeightResponse) {
-        for attestations in att_resp.attestations {
-            let state_map = self.0.entry(attestations.height).or_default();
-            state_map
-                .entry(State::from_slice(&attestations.data))
-                .or_default()
-                .push((
-                    Signature::from_slice(&attestations.signature),
-                    Pubkey::from_slice(&att_resp.pubkey),
-                ));
+    pub fn insert(&mut self, attestation: Attestation) {
+        if let Err(e) = attestation.validate() {
```

Comment:
Why do we ignore errors here? Seems like the caller should be the one responsible for deciding if the errors should be swallowed or not?